sync-team/src/crates_io/api.rs

```diff
@@ -0,0 +1,175 @@
+    }
+
+    /// Create a new trusted publishing configuration for a given crate.
+    pub(crate) fn create_trusted_publishing_github_config(
+        &self,
+        config: &TrustedPublishingConfig,
+    ) -> anyhow::Result<()> {
+        debug!(
+            "Creating trusted publishing config for '{}' in repo '{}/{}', workflow file '{}' and environment '{}'",
+            config.krate.0,
+            config.repo_name,
+            config.repo_name,
+            config.workflow_file,
+            config.environment.as_deref().unwrap_or("(none)")
+        );
+
+        if self.dry_run {
+            return Ok(());
+        }
+
+        #[derive(serde::Serialize)]
+        struct TrustedPublishingGitHubConfigCreate<'a> {
+            repository_owner: &'a str,
+            repository_name: &'a str,
+            #[serde(rename = "crate")]
+            krate: &'a str,
+            workflow_filename: &'a str,
+            environment: Option<&'a str>,
+        }
+
+        #[derive(serde::Serialize)]
+        struct CreateTrustedPublishing<'a> {
+            github_config: TrustedPublishingGitHubConfigCreate<'a>,
+        }
+
+        let request = CreateTrustedPublishing {
+            github_config: TrustedPublishingGitHubConfigCreate {
+                repository_owner: &config.repo_org,
+                repository_name: &config.repo_name,
+                krate: &config.krate.0,
+                workflow_filename: &config.workflow_file,
+                environment: config.environment.as_deref(),
+            },
+        };
+
+        self.req(
+            reqwest::Method::POST,
+            "/trusted_publishing/github_configs",
+            Some(&request),
+        )?
+        .error_for_status()
+        .with_context(|| anyhow!("Cannot created trusted publishing config {config:?}"))?;
+
+        Ok(())
+    }
+
+    /// Delete a trusted publishing configuration with the given ID.
+    pub(crate) fn delete_trusted_publishing_github_config(
+        &self,
+        id: TrustedPublishingId,
+    ) -> anyhow::Result<()> {
+        debug!("Deleting trusted publishing with ID {id}");
+
+        if !self.dry_run {
+            self.req::<()>(
+                reqwest::Method::DELETE,
+                &format!("/trusted_publishing/github_configs/{}", id.0),
+                None,
+            )?
+            .error_for_status()
+            .with_context(|| anyhow!("Cannot delete trusted publishing config with ID {id}"))?;
+        }
+
+        Ok(())
+    }
+
+    /// Perform a request against the crates.io API
+    fn req<T: Serialize>(
+        &self,
+        method: reqwest::Method,
+        path: &str,
+        data: Option<&T>,
+    ) -> anyhow::Result<reqwest::blocking::Response> {
+        let mut req = self
+            .client
+            .request(method, format!("{CRATES_IO_BASE_URL}{path}"))
+            .header(
+                "Authorization",
+                format!("Bearer {}", self.token.expose_secret()),
+            );
```

Comment:
```suggestion
            .bearer_auth(self.token.expose_secret());
```

I think this should be sufficient, but I might be wrong 😅